client/v2/algod/getApplicationBoxByName.go

```diff
@@ -0,0 +1,45 @@
+package algod
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/common"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+)
+
+// GetApplicationBoxByNameParams contains all of the query parameters for url serialization.
+type GetApplicationBoxByNameParams struct {
+
+	// Name a box name, in the goal app call arg form 'encoding:value'. For ints, use
+	// the form 'int:1234'. For raw bytes, use the form 'b64:A=='. For printable
+	// strings, use the form 'str:hello'. For addresses, use the form 'addr:XYZ...'.
+	Name string `url:"name,omitempty"`
```

Comment:
is `omitempty` necessary if this is a mandatory parameter?

Reply:
@algochoi It's a fair question - let me know if my response helps.  Happy to discuss live too.

My 2c:
* There might also be a question if `omitemty` is _ever_ necessary.  Certainly, here it seems a bit strange.
* For the use case at-hand:  It's generated code.  And my unverified guess is _generator_ does not disambiguate required vs optional query string parameters.